packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/useUploadView.ts

```diff
@@ -1,73 +1,47 @@
 import React from 'react';
-import { isUndefined } from '@aws-amplify/ui';
 
 import type { UploadHandlerData } from '../../../actions';
-import type { FileItems } from '../../../files';
 import { useFiles } from '../../../files';
 import { useStore } from '../../../store';
 import type { Task } from '../../../tasks';
 import { useAction } from '../../../useAction';
-import { isFileTooBig } from '../../../validators';
 
-import type { UploadViewState, UseUploadViewOptions } from './types';
 import { DEFAULT_OVERWRITE_ENABLED } from './constants';
-
-interface FilesData {
-  invalidFiles: FileItems | undefined;
-  validFiles: FileItems | undefined;
-  data: UploadHandlerData[];
-}
+import type { UploadViewState, UseUploadViewOptions } from './types';
 
 export const useUploadView = (
   options?: UseUploadViewOptions
 ): UploadViewState => {
   const { onExit: _onExit } = options ?? {};
 
   const [{ location }, storeDispatch] = useStore();
-  const [files, filesDispatch] = useFiles();
+  const [{ items: fileItems, invalidFiles }, filesDispatch] = useFiles();
   const { current } = location;
 
   const [isOverwritingEnabled, setIsOverwritingEnabled] = React.useState(
     DEFAULT_OVERWRITE_ENABLED
   );
 
-  const filesData = React.useMemo(
+  const uploadData = React.useMemo(
     () =>
-      (files ?? [])?.reduce(
-        (curr: FilesData, item) => {
-          if (isFileTooBig(item.file)) {
-            curr.invalidFiles = isUndefined(curr.invalidFiles)
-              ? [item]
-              : curr.invalidFiles.concat(item);
-          } else {
-            curr.validFiles = isUndefined(curr.validFiles)
-              ? [item]
-              : curr.validFiles.concat(item);
-
-            const parsedFileItem = {
-              ...item,
-              key: `${location.key}${item.key}`,
-            };
-
-            curr.data = curr.data.concat({
-              ...parsedFileItem,
-              preventOverwrite: !isOverwritingEnabled,
-            });
-          }
-
-          return curr;
-        },
-        { invalidFiles: undefined, validFiles: undefined, data: [] }
-      ),
-    [files, isOverwritingEnabled, location.key]
+      (fileItems ?? []).reduce((curr: UploadHandlerData[], item) => {
+        const parsedFileItem = {
+          ...item,
+          key: `${location.key}${item.key}`,
+        };
+
+        return curr.concat({
+          ...parsedFileItem,
+          preventOverwrite: !isOverwritingEnabled,
+        });
+      }, []),
```

Comment:
Since the initial value is a new array can use `.push` as there is no need to provide a new reference on each mutation. There's also no need to create `parsedFileItem` and add `preventOverwrite` in multiple operations. Can simplify to:
```ts
  (fileItems ?? []).reduce(
    (curr: UploadHandlerData[], item) =>
      curr.push({
        ...item,
        key: `${location.key}${item.key}`,
        preventOverwrite: !isOverwritingEnabled,
      }),
    []
  )
```